docs/source/pages/overview.rst

```diff
@@ -292,6 +292,14 @@ inside your LightningModule
     have the same call signature. If this is not the case, input that should be
     given to different metrics can given as keyword arguments to the collection.
 
+An additional advantage of using the ``MetricCollection`` object is that it will
+automatically try to reduce the computations needed by finding groups of metrics
+that share the same underlying metric state. If such metrics are found only one
+of them are actually updated and the updated state will be broadcastet to the rest
+of the metrics. In the example above, this will lead to a 2x-3x lower computational
+cost compared to disabling this feature. See the `compute_groups` argument
+for more info.
+
```

Comment:
maybe add some examples of such groups, so for classification, it is quite natural, but it may be hard for audio...